### ISSUE TITLE
AZURE_* environment variables

### ISSUE THREAD

**Issue by ihnorton** [Azure/azure-storage-cpplite#74]
Please consider supporting the `AZURE_*` environment variables like the [azure CLI](https://docs.microsoft.com/en-us/cli/azure/storage/account?view=azure-cli-latest), especially:
- `AZURE_STORAGE_ACCOUNT`
- `AZURE_STORAGE_KEY`

For testing it would also be useful to be able to configure `AZURE_STORAGE_ENDPOINT` (not supported by the CLI, as far as I can tell).

**Comment by JinmingHu-MSFT** [comment]
Cpplite is different from Azure-CLI, which itself is an executable. But cpplite is a library. I'm not sure whether it's a good idea to handle these environment variables in a library. Maybe these account name/key and endpoint information should loaded from environment variables by the executable and passed to this library. Anyway, we will evaluate this feature request.

**Comment by ihnorton** [comment]
>  Maybe these account name/key and endpoint information should loaded from environment variables by the executable and passed to this library.

Yes, that is what we are doing for now. Just to note that the AWS S3 C++ SDK does fall back to environment variables, which is useful for several situations. Thanks for evaluating this feature!

**Comment by JinmingHu-MSFT** [comment]
@ihnorton Thanks for your feedback, we'll add this feature to our backlog, and maybe support it in future release.
